kotlinx-coroutines-core/common/src/flow/operators/Emitters.kt

```diff
@@ -55,7 +55,12 @@ internal inline fun <T, R> Flow<T>.unsafeTransform(
 }
 
 /**
- * Invokes the given [action] when this flow starts to be collected.
+ * Returns a flow that invokes the given [action] **before** this flow starts to be collected.
+ *
+ * The [action] is called before the upstream flow is started, so if it is used with a [SharedFlow]
+ * there is **no guarantee** that emissions to the upstream flow that happen inside or immediately
```

Comment:
`to the upstream` -> `from the upstream`?

Reply:
Yes